### PR TITLE
KIALI-3114 Refresh PfTitle when props change

### PR DESCRIPTION
**Pull request by lucasponce**
Fix bug on PfTitle as state is not properly refreshed when props change.

### CODE REVIEW

File: src/components/Pf/PfTitle.tsx [hunk 1]
```diff
@@ -68,15 +79,30 @@ class PfTitle extends React.Component<PfTitleProps, PfTitleState> {
...
       type: type,
       name: name,
       graphType: graphType,
       icon: icon,
       cytoscapeGraph: cytoscapeGraph.build()
-    };
+    });
+  }
+
+  componentDidMount(): void {
```
Comment (cfcosta) [inline src/components/Pf/PfTitle.tsx]: I don't think we historically have added `void` type annotations for functions, but that's just a nitpick.

File: src/components/Pf/PfTitle.tsx [hunk 2]
```diff
@@ -68,15 +79,30 @@ class PfTitle extends React.Component<PfTitleProps, PfTitleState> {
...
+    this.doRefresh();
+  }
+
+  componentDidUpdate(prevProps: Readonly<PfTitleProps>): void {
```
Comment (cfcosta) [inline src/components/Pf/PfTitle.tsx]: Same here.

File: src/components/Pf/PfTitle.tsx (hunk 1 above)
Comment (lucasponce) [reply src/components/Pf/PfTitle.tsx]: We have them in other locations.

Comment (cfcosta) [reply src/components/Pf/PfTitle.tsx]: Np then :)

**Comment by lucasponce** [comment]
@hhovsepy let me know if the issue is gone or if there is anything else not covered by the fix.

**Pull request merged**